controllers/drplacementcontrol.go

```diff
@@ -535,20 +535,25 @@ func (d *DRPCInstance) checkMetroFailoverPrerequisites(curHomeCluster string) (b
 func (d *DRPCInstance) checkRegionalFailoverPrerequisites() bool {
 	d.setProgression(rmn.ProgressionWaitForStorageMaintenanceActivation)
 
-	if required, activationsRequired := requiresRegionalFailoverPrerequisites(
-		d.ctx,
-		d.reconciler.APIReader,
-		rmnutil.DRPolicyS3Profiles(d.drPolicy, d.drClusters).List(),
-		d.instance.GetName(), d.instance.GetNamespace(),
-		d.vrgs, d.instance.Spec.FailoverCluster,
-		d.reconciler.ObjStoreGetter, d.log); required {
-		for _, drCluster := range d.drClusters {
-			if drCluster.Name != d.instance.Spec.FailoverCluster {
-				continue
-			}
+	for _, drCluster := range d.drClusters {
+		if drCluster.Name != d.instance.Spec.FailoverCluster {
+			continue
+		}
+
+		// we want to work with failover cluster only, because the previous primary cluster might be unreachable
+		s3Clusters := []rmn.DRCluster{drCluster}
 
+		if required, activationsRequired := requiresRegionalFailoverPrerequisites(
+			d.ctx,
+			d.reconciler.APIReader,
+			rmnutil.DRPolicyS3Profiles(d.drPolicy, s3Clusters).List(),
```

Comment:
We could just use `[]string{drCluster.Spec.S3ProfileName}` here instead of crafting the array and then calling the function. e.g: https://github.com/RamenDR/ramen/blob/ca7af2de2a15cec8d3b59d21982fd8b7ff5ab26b/controllers/drcluster_mmode.go#L76

Reply:
Done